### PR TITLE
Bugfix/EUI-2276 Casefield showing summary screen when complex type field hidden

### PR DESCRIPTION
**Pull request by kuda1992**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2276?filter=-1

### Change description ###
Complex `Casefield` labels are showing in CYA page. To remove labels on CYA page I have introduced  `CcdCYAPageLabelFilterPipe` pipe which will remove all the `CaseField` children labels.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by connorpgpmcelroy — APPROVED** [review]

**Review by leepavlouhmcts — DISMISSED** [review]

File: src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+      if (caseField.field_type.complex_fields && caseField.field_type.complex_fields.length) {
+        caseField.field_type.complex_fields = this.removeLabels(caseField.field_type.complex_fields);
+      }
+
+      return caseField;
+    });
+  }
+
+  private removeLabels = (complexFields: CaseField[]): CaseField[] =>
```
Comment (udaydenduluri33) [inline src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts:20]: Should this method be called removeLabels? It's not removing anything rather applying a filter. I

**Review by udaydenduluri33 — APPROVED** [review]
Approved with 1 minor suggestion

File: src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts (hunk 1 above)
Comment (kuda1992) [reply src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts:20]: Well it's taking a list of `CaseFields[]` and removing all CaseFields of type Label.

Comment (udaydenduluri33) [reply src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts:20]: I understand that. But, the private method removeLabels is not removing it. Rather the result of the method removeLabels is doing that. So, the method could be called getNonLabelComplexFieldType? Agree?

Comment (kuda1992) [reply src/shared/components/palette/complex/ccd-cyapage-label-filter.pipe.ts:20]: Oh I see what you mean. What you're saying makes sense I will change the method name. Agreed

**Review by adnanakgun — DISMISSED** [review]

File: src/shared/components/case-editor/case-edit-page/case-edit-page.html [hunk 2]
```diff
@@ -22,12 +22,12 @@ <h2 class="govuk-error-summary__title" id="error-summary-title">
   </h2>
   <div *ngFor="let validationError of validationErrors" class="govuk-error-summary__body">
     <ul class="govuk-list govuk-error-summary__list">
-      <li> 
+      <li>
```
Comment (udaydenduluri33) [inline src/shared/components/case-editor/case-edit-page/case-edit-page.html:25]: please remove the formatting changes on all your files

File: src/shared/components/case-editor/case-edit-submit/case-edit-submit.html [hunk 3]
```diff
@@ -43,7 +43,10 @@ <h2 class="heading-h2">Check your answers</h2>
       <tbody>
       <ng-container *ngFor="let page of this.wizard.pages">
         <ng-container *ngIf="isShown(page)">
-          <ng-container *ngFor="let field of page | ccdPageFields:editForm.controls['data'] | ccdReadFieldsFilter: false :undefined :true :editForm.controls['data']">
+          <ng-container *ngFor="let field of page
+          | ccdPageFields: editForm.controls['data']
```
Comment (udaydenduluri33) [inline src/shared/components/case-editor/case-edit-submit/case-edit-submit.html:47]: please remove the formatting changes on all your files

File: src/shared/components/case-editor/case-edit-submit/case-edit-submit.html [hunk 4]
```diff
@@ -69,21 +72,21 @@ <h2 class="heading-h2">Check your answers</h2>
...
-                </td>
-              </tr>
-              <tr *ngSwitchCase="false" class="compound-field" ccdLabelSubstitutor [caseField]="field" [formGroup]="editForm.controls['data']" [contextFields]="contextFields">
-                <td colspan="2">
-                  <ccd-field-read [formGroup]="editForm.controls['data']" [caseField]="summaryCaseField(field)" [caseFields]="contextFields"></ccd-field-read>
-                </td>
-              </tr>
-            </ng-container>
-          </ng-container>
+      <ng-container *ngFor="let field of this.showSummaryFields">
```
Comment (udaydenduluri33) [inline src/shared/components/case-editor/case-edit-submit/case-edit-submit.html:75]: please remove the formatting changes on all your files

**Review by udaydenduluri33 — APPROVED** [review]
@kuda1992 Thanks for making the formatting changes

**Review by adnanakgun — APPROVED** [review]

**Pull request merged**